### PR TITLE
Update `atmos workflow` command to allow restarting workflows from a named step. Update workflow docs

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Update `atmos workflow` command to allow restarting workflows from a named step
* Update workflow docs https://pr-300.atmos-docs.ue2.dev.plat.cloudposse.org/core-concepts/workflows/#executing-workflow-from-a-named-step

## why
Useful when you want to restart a workflow from a particular step.

Each workflow step can be given an arbitrary name (step's identifier) using the `name` attribute. For example:

```yaml
workflows:
  test-1:
    description: "Test workflow"
    steps:
      - command: echo Command 1
        name: step1
        type: shell
      - command: echo Command 2
        name: step2
        type: shell
      - command: echo Command 3
        name: step3
        type: shell
      - command: echo Command 4
        type: shell
```

The step's name can be used in the `--from-step` command-line flag to start the workflow execution from the step.

For example, the following command will skip the first two steps and will start executing the workflow from `step3`:

```console
atmos workflow test-1 -f workflow1 --from-step step3
```

This is useful when you want to restart the workflow from a particular step.

For example:

- You run the workflow first time with the command `atmos workflow test-1 -f workflow1`
- `step1` and `step2` succeed, but `step3` fails
- You fix the issue with the `step3` command
- You don't want to execute `step1` and `step2` again (to not spend time on it, or if they are
  not [idempotent](https://en.wikipedia.org/wiki/Idempotence))
- You run the command `atmos workflow test-1 -f workflow1 --from-step step3` to restart the workflow from `step3`

## related
* Closes https://github.com/cloudposse/atmos/issues/271

## test

```shell
atmos workflow test-1 -f workflow1 --from-step step3
```

```console
Executing the workflow 'test-1' from 'examples/complete/stacks/workflows/workflow1.yaml'

description: Test workflow
steps:
- name: step1
  command: echo Command 1
  type: shell
- name: step2
  command: echo Command 2
  type: shell
- name: step3
  command: echo Command 3
  type: shell
- command: echo Command 4
  type: shell
- command: echo Command 5
  type: shell

Executing workflow step: echo Command 3

Executing command:
echo Command 3
Command 3

Executing workflow step: echo Command 4

Executing command:
echo Command 4
Command 4

Executing workflow step: echo Command 5

Executing command:
echo Command 5
Command 5
```

### CODE REVIEW

**Comment by osterman** [comment]
This closes #271

File: pkg/config/schema.go [hunk 1]
```diff
@@ -143,22 +143,27 @@ type ConfigAndStacksInfo struct {
 	AtmosBasePath             string
 }
 
+// Workflows
+
 type WorkflowStep struct {
+	Name    string `yaml:"name,omitempty" json:"name,omitempty" mapstructure:"name"`
```
Comment (osterman) [inline pkg/config/schema.go:149]: Can we  by default generate a friendly name, like `step1`, or `step2` based on the index in the list?

File: website/docs/core-concepts/workflows/workflows.md [hunk 2]
```diff
@@ -142,7 +145,10 @@ where:
 Each step is configured using the following attributes:
 
 - `command` - the command to execute. Can be either an Atmos [CLI command](/category/commands-1) (without the `atmos` binary name in front of it,
-  e.g. `command: terraform apply vpc`), or a shell script. The type of the command is specified by the `type` attribute
+  for example `command: terraform apply vpc`), or a shell script. The type of the command is specified by the `type` attribute
+
+- `name` - step name (optional). It's used to find the first step from which to start executing the workflow when the command-line flag `--from-step`
+  is specified
```
Comment (osterman) [inline website/docs/core-concepts/workflows/workflows.md:151]:
```suggestion
- `name` - step name (optional). It's used to find the first step from which to start executing the workflow when the command-line flag `--from-step`
  is specified. If the `name` is omitted, a friendly name will be generated for you consisting of a prefix of `name` and followed by the index of the step.
```

File: website/docs/core-concepts/workflows/workflows.md [hunk 3]
```diff
@@ -148,7 +148,7 @@ Each step is configured using the following attributes:
   for example `command: terraform apply vpc`), or a shell script. The type of the command is specified by the `type` attribute
 
 - `name` - step name (optional). It's used to find the first step from which to start executing the workflow when the command-line flag `--from-step`
-  is specified
+  is specified. If the `name` is omitted, a friendly name will be generated for you consisting of a prefix of `name` and followed by the index of the step.
```
Comment (osterman) [inline website/docs/core-concepts/workflows/workflows.md:151]: I meant to suggest a prefix of `step` (typo)

**Review by dudymas — APPROVED** [review]
Looks great!
thanks
image

**Review by osterman — APPROVED** [review]

**Pull request merged**